pkg/scaling/resolver/scale_resolvers.go

```diff
@@ -597,6 +642,170 @@ func resolveAuthSecret(ctx context.Context, client client.Client, logger logr.Lo
 	return string(result)
 }
 
+func resolveBoundServiceAccountToken(ctx context.Context, client client.Client, logger logr.Logger, namespace string, bsat *kedav1alpha1.BoundServiceAccountToken, triggerAuthRef *kedav1alpha1.AuthenticationRef, acs *authentication.AuthClientSet) string {
+	serviceAccountName, expiry := bsat.ServiceAccountName, bsat.Expiry
+	if serviceAccountName == "" {
+		logger.Error(fmt.Errorf("error trying to get token"), "serviceAccountName is required")
+		return ""
+	}
+	var err error
+	expirySeconds := ptr.Int64(3600)
+	if expiry != "" {
+		duration, err := time.ParseDuration(expiry)
+		if err != nil {
+			logger.Error(err, "error trying to parse expiry duration", "expiry", expiry)
+			return ""
+		}
+		// convert duration to seconds
+		expirySeconds = ptr.Int64(int64(duration.Seconds()))
+	}
+
+	triggerAuth, _, err := getTriggerAuth(ctx, client, triggerAuthRef, namespace)
+	if err != nil {
+		logger.Error(err, "error trying to get [cluster]triggerAuth", "TriggerAuth.Namespace", namespace, "TriggerAuth.Name", triggerAuthRef.Name)
+		return ""
+	}
+	currentAnnotations := triggerAuth.GetAnnotations()
+	encodedToken := currentAnnotations["keda-serviceAccountToken"]
+
+	// check if service account exists in the namespace
+	serviceAccount := &corev1.ServiceAccount{}
+	err = client.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: namespace}, serviceAccount)
+	if err != nil {
+		logger.Error(err, "error trying to get service account from namespace", "ServiceAccount.Namespace", namespace, "ServiceAccount.Name", serviceAccountName)
+		return ""
+	}
+
+	// check if token is already referenced in the TriggerAuthentication annotation
+	if encodedToken != "" {
+		tokenValid := checkTokenValidity(ctx, logger, encodedToken, expirySeconds, acs)
+		switch tokenValid {
+		case tokenStatusInvalid:
+			// token is invalid, or if more than 50% of the token's expiry has passed, create new token
+			return generateAndAnnotateNewToken(ctx, client, logger, serviceAccountName, namespace, expirySeconds, triggerAuth, acs)
+		case tokenStatusValid:
+			return encodedToken
+		default:
+			// tokenStatusUnknown
+			return ""
+		}
+	} else {
+		// token doesn't exist; create new token and embed it in the the TriggerAuth
+		logger.Info("Token doesn't exist; creating new token and embedding it in the triggerauth", "ServiceAccount.Namespace", namespace, "ServiceAccount.Name", serviceAccountName)
+		return generateAndAnnotateNewToken(ctx, client, logger, serviceAccountName, namespace, expirySeconds, triggerAuth, acs)
+	}
+}
```

Comment:
Currently, KEDA only re-executes the resolver in case of scaler creation. It means that the expiration of the token won't be checked until it has expired, triggering the error in the scaler and re-evaluating this code.
About storing the token in the TriggerAuthentication, I don't like this approach too much as it can be exposed just accessing the TriggerAuthentication CR, which isn't a sensitive API. 
Considering these 2 things together, I'd not store the token anywhere and just request a fresh token each time, storing it in the memory of the scaler. It solves the storing issue and removed the verification code (which won't be executed until the token has already expired in the common scenario)

Reply:
Okay, make sense. Then should users still be able to embed an "expiry"? I would imagine a failing query every expiry period might be annoying to some users for availability reasons, but it might help if it fails at expected intervals?

Reply:
Can we try to handle the expired token? The we can try to refresh the token and fail -> error if we are not able to do that.

Reply:
Does my current implementation in https://github.com/kedacore/keda/pull/6272/commits/44d840cc3fdad1f6bc4181daaa877b147d8e6600 make sense then?